### PR TITLE
Updates react to 154, electron to 147

### PR DESCRIPTION
**Pull request by chrisnojima**
Some issues with this upgrade

- [x] React needs react-tap-event-plugin to be updated, but the older version of material-ui has a peer dependency on the older version. I don't want to touch material-ui at this point so instead i have a postinstall script which removes the peer deps from the installed material-ui. This doesn't affect anything other than running shrinkwrap, else it'll give you a warning and error out
- [x] Added some flow ignores to some node_modules/tests which have issues
- [x] Fixes some small lint issues w/ new eslint

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

**Comment by keybase-ci-visdiff** [comment]
The commits 4022693e454e...61cd827b2f85 introduce visual changes on linux. <details><summary>:mag_right: 10 removed</summary>

 * Removed: **Settings_Nav-Normal_Good_Banner** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479761904268-4022693e454e-61cd827b2f85-linux/Settings_Nav-Normal_Good_Banner-4022693e454e.png)
 * Removed: **Settings_Nav-Normal_Bad_Banner** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479761904268-4022693e454e-61cd827b2f85-linux/Settings_Nav-Normal_Bad_Banner-4022693e454e.png)
 * Removed: **Settings_Nav-Normal** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479761904268-4022693e454e-61cd827b2f85-linux/Settings_Nav-Normal-4022693e454e.png)
 * Removed: **Search-Searching** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479761904268-4022693e454e-61cd827b2f85-linux/Search-Searching-4022693e454e.png)
 * Removed: **Search-Group_non_user_Out_of_invites** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479761904268-4022693e454e-61cd827b2f85-linux/Search-Group_non_user_Out_of_invites-4022693e454e.png)
 * Removed: **Search-Group_non_user_No_Avatar** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479761904268-4022693e454e-61cd827b2f85-linux/Search-Group_non_user_No_Avatar-4022693e454e.png)
 * Removed: **Search-Group_non_user_Has_Invite** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479761904268-4022693e454e-61cd827b2f85-linux/Search-Group_non_user_Has_Invite-4022693e454e.png)
 * Removed: **Search-Group_non_user** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479761904268-4022693e454e-61cd827b2f85-linux/Search-Group_non_user-4022693e454e.png)
 * Removed: **Search-Group** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479761904268-4022693e454e-61cd827b2f85-linux/Search-Group-4022693e454e.png)
 * Removed: **Search-Chat_enabled** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479761904268-4022693e454e-61cd827b2f85-linux/Search-Chat_enabled-4022693e454e.png)
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 4022693e454e...61cd827b2f85 introduce visual changes on linux. <details><summary>:mag_right: 10 removed</summary>

 * Removed: **Settings_Nav-Normal_Good_Banner** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479766617057-4022693e454e-61cd827b2f85-linux/Settings_Nav-Normal_Good_Banner-4022693e454e.png)
 * Removed: **Settings_Nav-Normal_Bad_Banner** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479766617057-4022693e454e-61cd827b2f85-linux/Settings_Nav-Normal_Bad_Banner-4022693e454e.png)
 * Removed: **Settings_Nav-Normal** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479766617057-4022693e454e-61cd827b2f85-linux/Settings_Nav-Normal-4022693e454e.png)
 * Removed: **Search-Searching** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479766617057-4022693e454e-61cd827b2f85-linux/Search-Searching-4022693e454e.png)
 * Removed: **Search-Group_non_user_Out_of_invites** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479766617057-4022693e454e-61cd827b2f85-linux/Search-Group_non_user_Out_of_invites-4022693e454e.png)
 * Removed: **Search-Group_non_user_No_Avatar** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479766617057-4022693e454e-61cd827b2f85-linux/Search-Group_non_user_No_Avatar-4022693e454e.png)
 * Removed: **Search-Group_non_user_Has_Invite** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479766617057-4022693e454e-61cd827b2f85-linux/Search-Group_non_user_Has_Invite-4022693e454e.png)
 * Removed: **Search-Group_non_user** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479766617057-4022693e454e-61cd827b2f85-linux/Search-Group_non_user-4022693e454e.png)
 * Removed: **Search-Group** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479766617057-4022693e454e-61cd827b2f85-linux/Search-Group-4022693e454e.png)
 * Removed: **Search-Chat_enabled** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479766617057-4022693e454e-61cd827b2f85-linux/Search-Chat_enabled-4022693e454e.png)
</summary>

**Comment by chrisnojima** [comment]
@keybase/react-hackers this updates the deps again to include electron 1.4.8

File: desktop/npm-helper.js [hunk 1]
```diff
@@ -202,6 +202,11 @@ function postInstall () {
   } else {
     exec("mkdir -p node_modules/msgpack; echo 'module.exports = null' > node_modules/msgpack/index.js; echo '{\"main\": \"index.js\"}' > node_modules/msgpack/package.json")
   }
+
```
Comment (gabriel) [inline desktop/npm-helper.js]: Maybe add short comment about what this is all about?

File: react-native/make-shim.js [hunk 2]
```diff
@@ -18,8 +18,10 @@ try {
   fs.writeFileSync(path.join(root, 'package.json'), `{
   "main": "index.js"
 }
-`) } catch (_) {}
+`)
+} catch (_) {}
```
Comment (gabriel) [inline react-native/make-shim.js]: Should we log error on all these try/catch failures? We weren't before so I guess not?

Comment (chrisnojima) [inline react-native/make-shim.js]: We don't check for existence and things like that which would be caught. I think in practice this thing never fails and if it did you're node_modules wouldn't work

File: desktop/npm-helper.js (hunk 1 above)
Comment (chrisnojima) [inline desktop/npm-helper.js]: done. in the file system refactor i actually changed this to use make-shim

**Pull request merged**